pints/_mcmc/__init__.py

```diff
@@ -300,22 +309,65 @@ class MCMCController(object):
     """
 
     def __init__(
-            self, log_pdf, chains, x0, sigma0=None, transform=None,
+            self, log_pdf, chains, x0=None, sigma0=None, transform=None,
             method=None):
 
         # Check function
         if not isinstance(log_pdf, pints.LogPDF):
             raise ValueError('Given function must extend pints.LogPDF')
 
+        # Get number of parameters
+        self._n_parameters = log_pdf.n_parameters()
+
+        # Check number of chains
+        self._n_chains = int(chains)
+        if self._n_chains < 1:
+            raise ValueError('Number of chains must be at least 1.')
+
+        # Get functions which can generate initial position or set initial
+        # position points if numerical values given
+        self._x0_isfunction = False
+        if x0 is None:
+            # use prior if one's available
+            if isinstance(log_pdf, pints.LogPosterior):
+                x0 = log_pdf.log_prior()
+            # do what Stan does and initialise all parameters on U(-2, 2) scale
+            else:
+                x0 = pints.UniformLogPrior([-2] * self._n_parameters,
+                                           [2] * self._n_parameters)
+        if isinstance(x0, pints.LogPrior):
+            init = [x0.sample() for i in range(chains)]
+            self._init_fn = x0
+            self._x0_isfunction = True
+        else:
+            if len(x0) != chains:
+                raise ValueError(
+                    'Number of initial positions must be equal to number of'
+                    ' chains.')
+            if not all([len(x) == self._n_parameters for x in x0]):
+                raise ValueError(
+                    'All initial positions must have the same dimension as the'
+                    ' given LogPDF.')
+            init = x0
+
         # Apply a transformation (if given). From this point onward the MCMC
         # sampler will see only the transformed search space and will know
         # nothing about the model parameter space.
         if transform is not None:
             # Convert log pdf
             log_pdf = transform.convert_log_pdf(log_pdf)
 
+        # Store transform for later detransformation: if using a transform, any
+        # parameters logged to the filesystem or printed to screen should be
+        # detransformed first!
+        self._transform = transform
```

Comment:
The transform code was handled first, so that transformed log_pdfs would be used throughout. With this re-ordering we sample initial points from the untransformed space?

Reply:
The init values are stored in a temp variable "init". It isn't stored to any object. Then a few lines down, they're converted to the correct space. These init values are needed for sigma0, so not sure how we could do the transform first.